src/searchers/beam_search.cpp

```diff
@@ -373,7 +373,11 @@ void BeamSearch::searchTopK(std::tuple<float *, int, int> &result) {
     // 4. add beam socre. Initialize -1e9 to all beams except the first one
     if (msgerSize > 1) {
         // Get the maximum value of each beam through all instance
-        float maxVal[batchSize * numBeams] = {std::numeric_limits<float>::lowest()};
+        float maxVal[batchSize * numBeams];
+        for (int i = 0; i < batchSize * numBeams; ++i) {
+            maxVal[i] = std::numeric_limits<float>::lowest();
```

Comment:
I guess the author intended to set all values to lowest float. good fix.

Reply:
Yes, original code could work, but this is c++20 standard. If we want to build w/ ICX, it should be c++17 standard.